File: backend/medbuddy/src/main/java/com/medbuddy/features/dashboard/DoctorDashboardResponse.java
package com.medbuddy.features.dashboard;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DoctorDashboardResponse {

    private String doctorName;
    private int todayAppointmentsCount;
    private int completedTodayCount;
    private NextPatientDto nextPatient;

    @Builder.Default
    private List<AppointmentSummaryDto> upcomingToday = List.of();
}

File: mobile/app/src/main/java/com/medbuddy/dto/DoctorDashboardResponse.kt
package com.medbuddy.dto

data class DoctorDashboardResponse(
    val doctorName: String,
    val todayAppointmentsCount: Int,
    val completedTodayCount: Int,
    val nextPatient: NextPatientDto? = null,
    val upcomingToday: List<AppointmentSummaryDto> = emptyList()
)

data class NextPatientDto(
    val appointmentId: Long? = null,
    val patientName: String,
    val appointmentTime: String,
    val reasonForVisit: String
)

data class AppointmentSummaryDto(
    val appointmentId: Long? = null,
    val patientName: String,
    val reasonForVisit: String,
    val appointmentTime: String
)